feat: tests for search-client-os project

yevhen-cap opened the pull request:
- connection tests for regular and async client
- added new project to pipeline

## Description

<!-- Describe the goal and purpose of this PR. -->

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

igpetrov commented on .github/workflows/zeebe-aws-os-dispatch-tests.yml:
```diff
@@ -17,14 +17,22 @@ jobs:
             maven-test-fork-count: 7
             runs-on: aws-core-8-default
+          - group: search-client-os
+            name: "Search Client OS Integration Tests"
```
```suggestion
            name: "Search Client AWS OS Integration Tests"
```

igpetrov commented on search/search-client-connect/src/test/java/io/camunda/search/connect/os/OpensearchConnectorIT.java:
```diff
@@ -0,0 +1,40 @@
+
+  @Test
+  void shouldCreateConnectionToAWSOpenSearchAndRetrieveData() throws Exception {
+    final var url = System.getProperty("camunda.it.opensearch.aws.url");
```
Let's make the property name `camunda.it.opensearch.aws.url` constant string

igpetrov commented on search/search-client-connect/src/test/java/io/camunda/search/connect/os/OpensearchConnectorIT.java:
```diff
@@ -0,0 +1,40 @@
+import io.camunda.search.connect.configuration.ConnectConfiguration;
+import org.junit.jupiter.api.Test;
+
+public class OpensearchConnectorIT {
```
Let's be more specific on naming.

```suggestion
public class AWSOpensearchConnectorIT {
```

igpetrov reviewed (changes requested):
LGTM, few minor changes requested

igpetrov reviewed (approved):
LGTM, thank you!

UPD: please don't forget to squash commits into one when merging to `main`

Commit:
feat: tests for search-client-os project

- connection tests for regular and async client
- added new project to pipeline
